### PR TITLE
Compare loss and some docs and adding another option for normalization in defaults.yaml

### PR DESCRIPTION
**Pull request by tianyu-z**

### CODE REVIEW

**Comment by tianyu-z** [comment]
Maybe this ugly paint can make the review easier. (Please notice that we are using **possible version 2** in the pic.)
image

File: omnigan/losses.py [hunk 1]
```diff
@@ -303,6 +306,15 @@ def __call__(self, input, target, mask):
         return torch.mean(torch.abs(torch.mul((input - target), 1 - mask)))
 
 
+class CompareLoss(nn.Module):
```
Comment (vict0rsch) [inline omnigan/losses.py:309]: I think CompareLoss is not explicit enough, something like `GroundIntersectionLoss` or something, it's ok if it's not generic it's really special for the ground in our use case

File: omnigan/losses.py [hunk 2]
```diff
@@ -466,7 +479,7 @@ def __call__(self, prediction, target):
 
 class ADVENTAdversarialLoss(nn.Module):
     """
-        TODO
+        The first and second arguement are tensor. The third arguement is a discriminator model.
```
Comment (vict0rsch) [inline omnigan/losses.py:482]: This is a class level docstring so it should explain what's the goal of the class itself, why does it exist, what does it do, in what context. The details (of arguments for instance) should be in method docstrings (like `__call__`'s)

File: omnigan/losses.py [hunk 3]
```diff
@@ -483,6 +496,11 @@ def __call__(self, prediction, target, discriminator):
 
 
 def multiDiscriminatorAdapter(d_out, opts):
+    """
+    Because the OmniDiscriminator does not directly return a tensor (but a list of tensor).
```
Comment (vict0rsch) [inline omnigan/losses.py:500]: Describe the expected type of inputs and there nature. Yous should use vscode extension autodocstring to help you do that (and type hints :) )

type hints are almost like type comments: 

```
def affine_square(a: float, b: Optional[float] = 0) -> float:
    return a ** 2 + b
```

https://kishstats.com/python/2019/01/07/python-type-hinting.html

**Review by vict0rsch — CHANGES_REQUESTED** [review]
Small changes, thanks for the PR it's cool :)

File: shared/trainer/defaults.yaml [hunk 4]
```diff
@@ -214,8 +219,8 @@ train:
       ent_var: 0.1 # the proportion of variance of entropy map in the entropy measure for a certain picture
       adv_main: 1.0 # the coefficient of the AdvEnt loss that minimize the entropy of the image by adversarial training
       adv_aux: 0.0 # the corresponding coefficient of the AdvEnt loss of second output
-      dis_main: 0.5 # the discriminator take care of the first output in the adversarial training
-      dis_aux: 0.5 # the discriminator take care of the second output in the adversarial training
+      dis_main: 1 # the discriminator take care of the first output in the adversarial training
```
Comment (alexrey88) [inline shared/trainer/defaults.yaml:222]: We're not using dis_main and dis_aux in the code right now right? And maybe we should get rid of ent_aux and adv_aux also since we're not using them?

Comment (vict0rsch) [reply shared/trainer/defaults.yaml:222]: Good point, if we're not using parameters we might as well clean them away

**Comment by vict0rsch** [comment]
what's up here?
